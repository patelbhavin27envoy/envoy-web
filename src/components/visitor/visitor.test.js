import React from 'react';
import renderer from 'react-test-renderer';

import { shallow, mount } from "enzyme";
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";

import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import { Visitor } from '../Visitor'

let visitors = [{
        firstName : 'John',
        lastName : 'Doe',
        notes : 'test first',
        signOutTime : new Date('02/09/2019').toDateString()
    },
    {
        firstName : 'Esteban',
        lastName : 'Arango',
        notes : 'Frisbee and Vegan food',
        signOutTime : new Date('04/24/2019').toDateString()
    },
    {
        firstName : 'Ryan',
        lastName : 'Labouve',
        notes : 'Everything about Oklahoma',
        signOutTime : null,
        signedOut: null,
        _id :122
    },
    {
        firstName : 'Esteban',
        lastName : 'Arango',
        notes : 'Frisbee and Vegan food',
        signedOut: true,
        signedIn : true,
        signOutTime : new Date('04/24/2019').toDateString(),
        _id: 123
    },
    {
        firstName : 'Esteban',
        lastName : 'Arango',
        notes : 'Frisbee and Vegan food',
        signedOut: true,
        signedIn : false,
        signOutTime : new Date('04/24/2019').toDateString()
    }];

// For Redux
Enzyme.configure({ adapter: new Adapter() });
const mockStore = configureMockStore();
const store = mockStore({
    venues : visitors
});


describe("Visitor Component :: ", () => {

    it("should render without throwing an error", () => {
        expect(
            renderer.create(
                <table>
                    <tbody>
                        <Visitor visitor={visitors[0]}/>
                    </tbody>
                </table>
            )
        ).not.toBe(null);
    });

    it("should render ", () => {

        // Act
        const component = mount(
            <table>
                <tbody>
                    <Visitor visitor={visitors[2]}/>
                </tbody>
            </table>
        );

        // Assert
        expect(component).toMatchSnapshot();
    });


    it("should make API call with signin and signout ", () => {

        // Act
        let component = renderer.create(
            <table>
                <tbody>
                    <Visitor visitor={visitors[3]}/>
                </tbody>
            </table>
        ).toJSON();
        // snapshot
        expect(component).toMatchSnapshot();


        // Act
        component = renderer.create(
            <table>
                <tbody>
                <Visitor visitor={visitors[4]}/>
                </tbody>
            </table>
        ).toJSON();
        // snapshot
        expect(component).toMatchSnapshot();
    });

    it("should make API call :: Success ", () => {

        // Act
        let component = renderer.create(
            <table>
                <tbody>
                <Visitor visitor={visitors[2]}/>
                </tbody>
            </table>
        ).toJSON();
        // snapshot
        expect(component).toMatchSnapshot();


        // define the response
        const mockSuccessResponse = {status : true, visitor : [visitors[2]]};

        // let's mock the service call
        const endPoint = 'http://localhost:8080/api/visitors/122';
        const mockJsonPromise = Promise.resolve(mockSuccessResponse);
        const mockFetchPromise = Promise.resolve({
            json: () => mockJsonPromise,
        });
        jest.spyOn(global, 'fetch').mockImplementation(() => { return mockFetchPromise }); // 4


        // Act
        component= shallow(
                <Visitor visitor={visitors[2]}/>
        );

        // Assert
        expect(component.state('loading')).toBe(false);
        expect(component.state('error')).toBe(false);

        component.find('.btn-smaller').simulate('click');
        expect(component.state('loading')).toBe(true);
        expect(component.state('error')).toBe(false);
    });


    it("should make API call :: Failure ", () => {

        // define the response
        const mockSuccessResponse = {};

        // let's mock the service call
        const endPoint = 'http://localhost:8080/api/visitors/123';
        const mockJsonPromise = Promise.resolve(mockSuccessResponse);
        const mockFetchPromise = Promise.resolve({
            json: () => mockJsonPromise,
        });
        jest.spyOn(global, 'fetch').mockImplementation(() => { return mockFetchPromise }); // 4


        // Act
        const component = shallow(
            <Visitor visitor={visitors[2]} />
        );

        // Assert
        expect(component.state('loading')).toBe(false);
        expect(component.state('error')).toBe(false);

        component.find('.btn-smaller').simulate('click');
        expect(component.state('loading')).toBe(true);
    });


    it("should make API call :: Failure-2 ", () => {

        // define the response
        // const mockSuccessResponse = {};

        // let's mock the service call
        const endPoint = 'http://localhost:8080/api/visitors/';
        const mockJsonPromise = Promise.resolve({data : { statue: false} });
        const mockFetchPromise = Promise.resolve({
            json: () =>  mockJsonPromise
        });
        jest.spyOn(global, 'fetch').mockImplementation(() => { return mockFetchPromise }); // 4


        // Act
        const component = shallow(
            <Visitor visitor={visitors[2]} />
        );

        // Assert
        expect(component.state('loading')).toBe(false);
        expect(component.state('error')).toBe(false);

        component.find('.btn-smaller').simulate('click');
        expect(component.state('loading')).toBe(true);
    });

});

