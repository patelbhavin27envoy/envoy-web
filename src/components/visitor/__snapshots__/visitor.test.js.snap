// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Visitor Component ::  should make API call :: Success  1`] = `
<table>
  <tbody>
    <tr>
      <td
        className="p-2 border-t border-grey-light font-mono text-xs one"
      >
         
      </td>
      <td
        className="p-2 border-t border-grey-light font-mono text-xs two"
      >
        Everything about Oklahoma
      </td>
      <td
        className="p-1 border-t border-grey-light font-mono text-xs three"
      >
        <button
          className="btn btn-smaller btn--outline"
          onClick={[Function]}
        >
          Sign out
        </button>
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`Visitor Component ::  should make API call with signin and signout  1`] = `
<table>
  <tbody>
    <tr>
      <td
        className="p-2 border-t border-grey-light font-mono text-xs one"
      >
         
      </td>
      <td
        className="p-2 border-t border-grey-light font-mono text-xs two"
      >
        Frisbee and Vegan food
      </td>
      <td
        className="p-1 border-t border-grey-light font-mono text-xs three"
      >
        Wed Apr 24 2019
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`Visitor Component ::  should make API call with signin and signout  2`] = `
<table>
  <tbody>
    <tr>
      <td
        className="p-2 border-t border-grey-light font-mono text-xs one"
      >
         
      </td>
      <td
        className="p-2 border-t border-grey-light font-mono text-xs two"
      >
        Frisbee and Vegan food
      </td>
      <td
        className="p-1 border-t border-grey-light font-mono text-xs three"
      >
        <button
          className="btn btn-smaller btn--outline"
          onClick={[Function]}
        >
          Sign out
        </button>
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`Visitor Component ::  should render  1`] = `ReactWrapper {}`;
